src/parser.rs

```diff
@@ -399,56 +402,107 @@ pub enum ReturnType {
     Exception,
 }
 
-#[derive(Clone)]
 struct ParseState<'e> {
-    /// Reference to the scripting `Engine`.
```

Comment:
I suggest leave them at three slashes... this style is consistent with the rest of the code.  If we eventually make this type public, then it automatically generates documentation.